.github/workflows/tests.yml

```diff
@@ -71,7 +78,58 @@ jobs:
         run: python -m pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
 
       - name: Run tests
-        run: python -m pytest tests --cov=pennylane $COVERAGE_FLAGS -m "not qcut and not qchem" -n auto
+        run: python -m pytest tests --cov=pennylane $COVERAGE_FLAGS -m "$INTERFACE" -n auto
+
+      - name: Adjust coverage file for Codecov
+        run: bash <(sed -i 's/filename=\"/filename=\"pennylane\//g' coverage.xml)
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          file: ./coverage.xml
+
+  math:

```

Comment:
It is true they have same configuration, but I think we should not mix them because they are clearly separated and is easier to read on the CI checks. But there is code duplication yes.